File: java2csharp/java/Samsung/Look.Pen/WritingBuddyEditTextActivity.java

package com.samsung.android.example.slookdemos;

import android.app.Activity;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;

import com.samsung.android.sdk.look.writingbuddy.SlookWritingBuddy;

public class WritingBuddyEditTextActivity extends Activity {

    private SlookWritingBuddy mWritingBuddy;

    private ImageView mOutputImageView;

    /* State */
    private boolean mIsEnabled = true;

    private boolean mEnableImage = false;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_writingbuddy_edittext);

        EditText editInput = (EditText) findViewById(R.id.edit_input);
        mOutputImageView = (ImageView) findViewById(R.id.image_output);

        mWritingBuddy = new SlookWritingBuddy(editInput);

        Button enableButton = (Button) findViewById(R.id.btn_enable);
        enableButton.setOnClickListener(new View.OnClickListener() {

            public void onClick(View v) {
                if (mIsEnabled) {
                    ((Button) v).setText("Enable");
                    mIsEnabled = false;
                } else {
                    ((Button) v).setText("Disable");
                    mIsEnabled = true;
                }
                mWritingBuddy.setEnabled(mIsEnabled);
            }
        });

        Button modeButton = (Button) findViewById(R.id.btn_enable_image);
        modeButton.setOnClickListener(new View.OnClickListener() {

            public void onClick(View v) {
                if (mEnableImage) {
                    mEnableImage = false;
                    mWritingBuddy.setImageWritingListener(null);
                    ((Button) v).setText("Enable Image Listener");
                } else {
                    mEnableImage = true;
                    mWritingBuddy
                            .setImageWritingListener(new SlookWritingBuddy.ImageWritingListener() {
                                public void onImageReceived(Bitmap arg0) {
                                    mOutputImageView.setBackground(new BitmapDrawable(
                                            getResources(), arg0));
                                }
                            });
                    ((Button) v).setText("Disable Image Listener");
                }
            }
        });

    }
}

File: java2csharp/csharp/Samsung/Look.Pen/WritingBuddyEditTextActivity.cs
namespace com.samsung.android.example.slookdemos
{

	using Activity = android.app.Activity;
	using Bitmap = android.graphics.Bitmap;
	using BitmapDrawable = android.graphics.drawable.BitmapDrawable;
	using Bundle = android.os.Bundle;
	using View = android.view.View;
	using Button = android.widget.Button;
	using EditText = android.widget.EditText;
	using ImageView = android.widget.ImageView;

	using SlookWritingBuddy = com.samsung.android.sdk.look.writingbuddy.SlookWritingBuddy;

	public class WritingBuddyEditTextActivity : Activity
	{

		private SlookWritingBuddy mWritingBuddy;

		private ImageView mOutputImageView;

		/* State */
		private bool mIsEnabled = true;

		private bool mEnableImage = false;

		public override void onCreate(Bundle savedInstanceState)
		{
			base.onCreate(savedInstanceState);
			ContentView = R.layout.activity_writingbuddy_edittext;

			EditText editInput = (EditText) findViewById(R.id.edit_input);
			mOutputImageView = (ImageView) findViewById(R.id.image_output);

			mWritingBuddy = new SlookWritingBuddy(editInput);

			Button enableButton = (Button) findViewById(R.id.btn_enable);
			enableButton.OnClickListener = new OnClickListenerAnonymousInnerClassHelper(this);

			Button modeButton = (Button) findViewById(R.id.btn_enable_image);
			modeButton.OnClickListener = new OnClickListenerAnonymousInnerClassHelper2(this);

		}

		private class OnClickListenerAnonymousInnerClassHelper : View.OnClickListener
		{
			private readonly WritingBuddyEditTextActivity outerInstance;

			public OnClickListenerAnonymousInnerClassHelper(WritingBuddyEditTextActivity outerInstance)
			{
				this.outerInstance = outerInstance;
			}


			public virtual void onClick(View v)
			{
				if (outerInstance.mIsEnabled)
				{
					((Button) v).Text = "Enable";
					outerInstance.mIsEnabled = false;
				}
				else
				{
					((Button) v).Text = "Disable";
					outerInstance.mIsEnabled = true;
				}
				outerInstance.mWritingBuddy.Enabled = outerInstance.mIsEnabled;
			}
		}

		private class OnClickListenerAnonymousInnerClassHelper2 : View.OnClickListener
		{
			private readonly WritingBuddyEditTextActivity outerInstance;

			public OnClickListenerAnonymousInnerClassHelper2(WritingBuddyEditTextActivity outerInstance)
			{
				this.outerInstance = outerInstance;
			}


			public virtual void onClick(View v)
			{
				if (outerInstance.mEnableImage)
				{
					outerInstance.mEnableImage = false;
					outerInstance.mWritingBuddy.ImageWritingListener = null;
					((Button) v).Text = "Enable Image Listener";
				}
				else
				{
					outerInstance.mEnableImage = true;
					outerInstance.mWritingBuddy.ImageWritingListener = new ImageWritingListenerAnonymousInnerClassHelper(this);
					((Button) v).Text = "Disable Image Listener";
				}
			}

			private class ImageWritingListenerAnonymousInnerClassHelper : SlookWritingBuddy.ImageWritingListener
			{
				private readonly OnClickListenerAnonymousInnerClassHelper2 outerInstance;

				public ImageWritingListenerAnonymousInnerClassHelper(OnClickListenerAnonymousInnerClassHelper2 outerInstance)
				{
					this.outerInstance = outerInstance;
				}

				public virtual void onImageReceived(Bitmap arg0)
				{
					outerInstance.outerInstance.mOutputImageView.Background = new BitmapDrawable(Resources, arg0);
				}
			}
		}
	}

}